src/Session.php

```diff
@@ -89,10 +95,67 @@ public function __unset(string $name): void
             throw new RuntimeException('Cannot alter session after it is closed');
         }
 
+        if (!isset($this->contents[$name])) {
+            \trigger_error('Undefined property: ' . self::class . '::$' . $name, \E_USER_WARNING);
+            return;
+        }
+
         $this->modified = true;
         unset($this->contents[$name]);
     }
 
+    public function offsetSet($name, $value): void
+    {
+        if (!$this->isInitialized()) {
+            throw new RuntimeException('Session not initialized');
+        }
+
+        if (!$this->writeable) {
+            throw new RuntimeException('Cannot alter session after it is closed');
+        }
+
+        $this->modified = true;
+        $this->contents[$name] = $value;
+    }
+
+    public function offsetExists($name): bool
+    {
+        if (!$this->isInitialized()) {
+            throw new RuntimeException('Session not initialized');
+        }
+
+        return isset($this->contents[$name]);
+    }
+
+    public function offsetUnset($name): void
+    {
+        if (!$this->isInitialized()) {
+            throw new RuntimeException('Session not initialized');
+        }
+
+        if (!$this->writeable) {
+            throw new RuntimeException('Cannot alter session after it is closed');
+        }
+
+        if (!isset($this->contents[$name])) {
+            \trigger_error('Undefined array key "' . $name . '"', \E_USER_WARNING);
+            return;
+        }
+
+        $this->modified = true;
+        unset($this->contents[$name]);
+    }
+
+    public function offsetGet($name): mixed
+    {
+        if (!isset($this->contents[$name])) {
```

Comment:
Not needed, PHP will trigger the error on `return $this->contents[$name]` when $name does not exist in $this->contents.